pkg/storegateway/bucket_store_metrics.go

```diff
@@ -176,5 +179,17 @@ func NewBucketStoreMetrics(reg prometheus.Registerer) *BucketStoreMetrics {
 
 	m.indexHeaderReaderMetrics = indexheader.NewReaderPoolMetrics(prometheus.WrapRegistererWithPrefix("cortex_bucket_store_", reg))
 
+	m.iteratorLoadDurations = promauto.With(reg).NewSummaryVec(prometheus.SummaryOpts{
```

Comment:
I would use an histogram for these new metrics. Summaries are not aggregatable and so they're more difficult to query at scale. Cardinality is not an issue because `iterator` is bounded (we known in advance how many we have).